### ISSUE TITLE
question on npy to png

### ISSUE THREAD

**Issue by AlpsCV** [SamsungLabs/unified-demosaicing#5]
Based on the updated code, I tried converting _gt.npy to _gt_viz.png using the generate_output_images function. The .npy file was loaded through BasicDataset, and within generate_output_images, I skipped adding PSNR to the filename. I didn’t modify any other part of your code. However, the generated PNG (1440x2160) appears lighter in color compared to Fig. 2 in your paper. Could you please advice this?

The extracted code is shown below for your reference.

NOTE: the metatdata was loaded successfully.
current_dir = os.path.dirname(os.path.abspath(__file__))
dng_path = os.path.join(current_dir, "DSC01201_PSMS.dng")
metadata = get_metadata(dng_path)

`
def convert_gt_npy_to_png(loader, device, out_imgs_dir=None):
    with tqdm(total=len(loader), desc='Validation round', unit='batch', leave=False) as pbar:
        for batch in loader:
            gt_ = batch['gt']
            gt = gt_[:, :, :, :]
            gt = gt.to(device=device, dtype=torch.float32)
            gt = output_rescaling_quantization(gt)
            generate_output_images(out_imgs_dir+"gt", gt, np.ones(0), batch["img_file"])
        pbar.update(1)
`

`
if __name__ == '__main__':
    args = get_args()
    args = set_job_id(args)
    gpus_chosen, device = get_device(args)
    run_name = "local_run"
    odir = os.path.join(args.odir, run_name)
    os.makedirs(odir, exist_ok=True)

    assert args.load, "You must specify the model to load via --load"
    # Full-size dataset test
    test_full_dataset = BasicDataset(args.fulldir, split="all_gt", pattern='single', remosaic=False)
    test_full_loader = DataLoader(test_full_dataset, batch_size=1, shuffle=False)
    convert_gt_npy_to_png(test_full_loader, device, out_imgs_dir=odir)
`

**Comment by tedlasai** [comment]
Figure 2 in our paper shows the DNG renderings directly. Unfortunately we are unable to show with this rendering as it is Adobe's rendering. So for all qual results we use the simple camera pipeline.

**Comment by AlpsCV** [comment]
Thanks for the quick reply and clarification. I have no question on this topic.
